dialogy/plugins/preprocess/text/calibration.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, List, Dict
+
+import math
+
+
+def predict_alternative(
+    alternative: Dict[str, Any], vectorizer: Any, classifier: Any
+) -> float:
+    vec_ = vectorizer.transform([alternative["transcript"]]).tolist()[0]
+    features = [
+        alternative["am_score"] / len(alternative["transcript"].split()),
+        alternative["lm_score"] / len(alternative["transcript"].split()),
+        alternative["transcript"].count("UNK"),
+        len(alternative["transcript"].split()),
+        alternative["am_score"] / math.log(1 + len(alternative["transcript"].split())),
+        alternative["lm_score"] / math.log(1 + len(alternative["transcript"].split())),
+        alternative["am_score"] / math.sqrt(len(alternative["transcript"].split())),
+        alternative["lm_score"] / math.sqrt(len(alternative["transcript"].split())),
+    ]
+    return classifier.predict(vec_ + features).tolist()[0]
+
+
+# == filter_asr_output ==
+def filter_asr_output(
+    utterances: List[List[Dict[str, Any]]],
+    threshold: float,
+    vectorizer: Any,
+    classifier: Any,
+) -> List[List[Dict[str, Any]]]:
+    """
+    .. _filter_asr_output:
+
+    Filter ASR output to return `good` alternatives.
+
+    This functions returns the good alternatives in the same format as the input.
+    The good alternatives are decided if the pWER (predicted WER) is less than
+    `threshold`.
+
+    .. ipython:: python
+
+        from dialogy.plugins.preprocess.text.calibration import filter_asr_output
+
+        utterances = [[{"transcript": "This is a sentence", "am_score":230, "lm_score":100}, \
+                            {"transcript": "This is another sentence", "am_score":230, "lm_score":100}]]
+        filter_asr_output(utterances)
+
+    :param utterances: A structure representing ASR output. We support only:
+
+        1. :code:`List[List[Dict[str, Any]]]`
+
+    :type utterances: List[List[Dict[str, Any]]]
+    :return: Good alternatives.
+    :rtype: List[List[Dict[str, Any]]]
+    :raises: `AssertionError` if utterance isn't in the desired format.
+    """
+
+    assert len(utterances) == 1, f"Got {len(utterances)} instead of 1"
```

Comment:
This (`len(utterance) > 1`) is valid for google asr. Might become true for our ASRs too.